### PR TITLE
Prepare drop constant keyword.

### PR DESCRIPTION
**Pull request by ekpyron**
Part of #2502.

Adjust tests, documentation, style guide, std contracts in preparation of #4090.

This is the non-breaking subset of #4090.

### CODE REVIEW

File: docs/grammar.txt [hunk 1]
```diff
@@ -58,7 +58,7 @@ ArrayTypeName = TypeName '[' Expression? ']'
 FunctionTypeName = 'function' FunctionTypeParameterList ( 'internal' | 'external' | StateMutability )*
                    ( 'returns' FunctionTypeParameterList )?
 StorageLocation = 'memory' | 'storage'
-StateMutability = 'pure' | 'constant' | 'view' | 'payable'
+StateMutability = 'pure' | 'view' | 'payable'
```
Comment (axic) [inline docs/grammar.txt]: This cannot be removed here.

File: docs/miscellaneous.rst [hunk 2]
```diff
@@ -400,7 +400,6 @@ Modifiers
 - ``view`` for functions: Disallows modification of state - this is not enforced yet.
 - ``payable`` for functions: Allows them to receive Ether together with a call.
 - ``constant`` for state variables: Disallows assignment (except initialisation), does not occupy storage slot.
-- ``constant`` for functions: Same as ``view``.
```
Comment (axic) [inline docs/miscellaneous.rst]: This cannot be removed here.

File: docs/types.rst [hunk 3]
```diff
@@ -380,7 +380,7 @@ be passed via and returned from external function calls.
 
 Function types are notated as follows::
 
-    function (<parameter types>) {internal|external} [pure|constant|view|payable] [returns (<return types>)]
+    function (<parameter types>) {internal|external} [pure|view|payable] [returns (<return types>)]
```
Comment (axic) [inline docs/types.rst]: This cannot be removed here.

File: docs/grammar.txt (hunk 1 above)
Comment (ekpyron) [reply docs/grammar.txt]: OK - the idea of this (and the one's below) was to already update the documentation *as if constant had already been removed* and only mentioning it's existence in a note, but you're probably right that it's better to postpone that until the breaking change.

**Review by axic — DISMISSED** [review]
Can you add a syntax test which deals with `constant` (will be removed as part of #4090)

**Review by axic — DISMISSED** [review]

**Comment by axic** [comment]
Rebased after #4104.

**Review by axic — DISMISSED** [review]

**Comment by axic** [comment]
@chriseth OK with this? (mostly the documentation changes)

**Comment by ekpyron** [comment]
@axic I added another commit that removes ``view`` from end-to-end-tests in cases where it will fail once the ``ViewPureChecker`` enforces ``view`` in 0.5.0.

**Review by chriseth — APPROVED** [review]

**Pull request merged**